### ISSUE TITLE
Deserialization: Explicit the keys that made the deserialization of an object fail

### ISSUE THREAD

**Issue by a-corni** [pasqal-io/Pulser#780]
Lately, a serialized Device had an error in its terms, the `p_false_pos` argument of the `default_noise_model` was written `p_False_pos`. This made finding the mistake very hard. Perhaps that, in the eventuality of a failing deserialization, we could try to check if the error is coming from an error in the keys of the dictionnary of the JSON-formatted object ?

**Comment by HGSilveri** [comment]
Indeed, the JSON validator error message is almost always useless but what you are proposing sounds like we would start doing part of its job.

Side note: This type of error only comes if someone is tinkering with the abstract representation manually, which should not happen...
